.github/workflows/test.yaml

```diff
@@ -42,3 +39,16 @@ jobs:
         run: cargo fmt --all -- --check
       - name: Run clippy
         run: cargo clippy --all --all-targets -- -D warnings
+  feature_powerset:
```

Comment:
Just a thought.  Would we want to add "needs: lint" here?  It would lengthen the total time spent on PR testing, since this wouldn't start until after linting is done, but it would avoid doing this heavyweight check on a PR that's just failing on format or clippy anyways.

Reply:
Hmmm, what do you think compared to https://stackoverflow.com/questions/74690423/how-can-i-use-github-actions-concurrency-to-cancel-other-jobs-if-any-of-them-f ?

Reply:
I tend to try to minimize my use of third party actions, since they're running potentially untrusted code in my CI pipeline with access to secrets, but that's above average on the paranoia level.  If I were the maintainer, I'd prefer the straight dependency to avoid adding the extra external action, but I won't pitch a fit if you/the maintainers prefer to go the other way.

Reply:
Fair, it's a trade off between "average" runtime usage, "worst-case" runtime usage, and additional outside dependencies. I don't have a strong stance here.